static/src/javascripts/lib/partition.ts

```diff
@@ -0,0 +1,13 @@
+const partition = <T>(
+	array: T[],
+	callback: (element: T, i: number, array: T[]) => boolean,
+): [T[], T[]] =>
+	array.reduce(
+		(result: [T[], T[]], e, i) => {
+			callback(e, i, array) ? result[0].push(e) : result[1].push(e);
```

Comment:
FYI, [Typescript `lib.es5.d.ts` will refer to this callback as the `predicate`](https://github.com/microsoft/TypeScript/blob/4ce902c5fa31d402be78709b3c44bc60405bad02/lib/lib.es5.d.ts#L4070)

Reply:
You're right I should call it 'predicate'. I did in the tests but not the impl!

Reply:
[MDN calls it a `callbackFn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#syntax), so you’re not wrong ;)